manifests/volume.pp

```diff
@@ -222,12 +222,24 @@
       }
 
       # did the options change?
-      $current_options = getvar("gluster_volume_${title}_options")
-      if $current_options {
-        $_current = sort( split($current_options, ',') )
-      } else {
-        $_current = []
+      if (defined('$::gluster_volume_options')){
```

Comment:
You can use https://github.com/puppetlabs/puppetlabs-stdlib#fact which returns undef on unknown facts. I think you can then even use `fact("gluster_volume_options.${title}")`